Bnijmeijer fileio 1 5

BNijmeijer opened the pull request.

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into finish/hole-tiles

maxmeijer42 committed:
Move animation from Player to SmoothlyMovingGameObject class

* The player will now also move smoothly if Tile.PerformAction is overridden
* The SmoothlyMovingGameObject class can also be used for other moving entities such as a Guard or an Editor

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into finish/hole-tiles

maxmeijer42 committed:
Naamgeving

BNijmeijer committed:
Updated FileIO.Save method

FileIO.Save should fully function now.

jensj12 commented on MeesGame/Gameplay/GameManagement/GameObject.cs:
```diff
@@ -3,11 +3,11 @@
-    protected bool visible;
+    public GameObject parent;
+    public Vector2 position, velocity;
+    public int layer;
```
Is it still needed to make everything public?

nessemis committed:
refactored ui (again) and renamed it GUI

nessemis committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into PlayingStateUI

nessemis committed:
FunctionalFutherUI

nessemis committed:
default asset doortile

maxmeijer42 committed:
Veranderingen in klassestructuur

maxmeijer42 committed:
Remove useless override

nessemis committed:
GUI -> UI

maxmeijer42 commented on MeesGame/Gameplay/Level/FileIO.cs:
```diff
@@ -10,29 +11,34 @@ class FileIO
-        /// <param name="ea"> Dummy variable. </param>
-        public void Save(object obj, EventArgs ea)
+        /// <param name="level"> The level being saved. </param>
+        /// <param name="filename"> The name of the file the level is being saved to. </param>
```
Parameter does not exist

BNijmeijer replied:
Fixed

maxmeijer42 commented on MeesGame/Gameplay/Level/FileIO.cs:
```diff
@@ -10,29 +11,34 @@ class FileIO
             if (saveFileDialog.ShowDialog() == DialogResult.OK)
             {
+                fileName = saveFileDialog.FileName;
                 Stream stream = new FileStream(fileName, FileMode.Create);
```
Catch exceptions

BNijmeijer replied:
Fixed

maxmeijer42 commented on MeesGame/Gameplay/Level/Level.cs:
```diff
@@ -58,7 +59,9 @@ public void ReadXml(XmlReader reader)
 
         public void WriteXml(XmlWriter writer)
```
This class is never serialized I believe, so it does not need to implement IXmlSerializable

maxmeijer42 reviewed (commented):
Haven't tested it yet, but it looks good.

jensj12 commented on MeesGame/MeesGame.csproj:
```diff
@@ -47,6 +47,7 @@
     <Compile Include="Gameplay\GameObjects\Tile.cs" />
     <Compile Include="Gameplay\GameObjects\TileField.cs" />
     <Compile Include="Gameplay\GameStates\PlayingLevelState.cs" />
+    <Compile Include="Gameplay\Level\FileIO.cs" />
```
FileIO.cs is already in the list. This generates an error.

maxmeijer42 committed:
Merge branch 'master' into character

Commit:
Made the tiles and a gameoverstate

maxmeijer42 committed:
Add EnterTile function to Tile class

* Function can be used by end tiles for example

Commit:
Changes according to comments

nessemis committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into PlayingStateUI

# Conflicts:
#	MeesGame/Gameplay/GameObjects/DoorTile.cs

nessemis committed:
incremental

nessemis committed:
modified editor code to allow for overlays

nessemis committed:
incremental

nessemis committed:
incremental

nessemis committed:
incremental

nessemis committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into PlayingStateUI

# Conflicts:
#	MeesGame/Gameplay/Level/Level.cs

nessemis committed:
finished

nessemis committed:
used cleanup extension

nessemis committed:
backspace to go back

nessemis committed:
refactored startlevelbutton

noiw displays generate random level when nothing is selected

LisanneW committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into Keys

BNijmeijer committed:
Added FileIO.Load method, added LevelData class, modified FileIO.Save method

The LevelData class contains all data the editor needs for saving and
loading levels. The FileIO.Save and FileIO.Load methods have been
modified so both SHOULD fully function now.

nessemis committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into PlayingStateUI

# Conflicts:
#	MeesGame/Gameplay/GameManagement/SpriteSheet.cs
#	MeesGame/Gameplay/GameObjects/DoorTile.cs

BNijmeijer commented:
I've overhauled the FileIO.Save and FileIO.Load methods, because the way I was trying to serialize everything was becoming a huge mess. Now, everything you'd need to create a level is contained in the LevelData class, and as such, both methods use this class in order to (de-)compose levels.

maxmeijer42 committed:
Make saving work

maxmeijer42 committed:
Merge remote-tracking branch 'origin/master' into character

Conflicts:
	MeesGame/Gameplay/GameObjects/Player.cs
	MeesGame/Gameplay/GameObjects/TileField.cs

maxmeijer42 committed:
Merge remote-tracking branch 'origin/master' into PlayingStateUI

Conflicts:
	MeesGame/Gameplay/GameManagement/SpriteSheet.cs
	MeesGame/Gameplay/GameObjects/DoorTile.cs

maxmeijer42 committed:
Refactor

nessemis committed:
very incremental. --CHANGES NOT YET IMPLEMENTED

maxmeijer42 committed:
Loading levels (includes changes from PlayingStateUI and FileIO1.5

maxmeijer42 committed:
Fix unused parameters

maxmeijer42 committed:
Add support for generating mazes

nessemis committed:
Merge remote-tracking branch 'origin/PlayingStateUI' into PlayingStateUI

# Conflicts:
#	MeesGame/Gameplay/GameManagement/SpriteSheet.cs
#	MeesGame/Gameplay/GameObjects/DoorTile.cs

Commit:
reverted MazeGenerator to master

Commit:
small changed

nessemis committed:
ran final cleaup

maxmeijer42 committed:
Merged finish/hole tiles in this branch

Also, I edited it so that it works :)

Conflicts:
	MeesGame/Gameplay/GameStates/LoadMenuState.cs
	MeesGame/Gameplay/Level/Level.cs
	MeesGame/MeesGame.csproj

maxmeijer42 committed:
Merge remote-tracking branch 'origin/PlayingStateUI' into loading-levels

Conflicts:
	MeesGame/Gameplay/GameStates/LoadMenuState.cs
	MeesGame/Gameplay/Level/EditorLevel.cs

Commit:
refactored a bit

Commit:
Merge remote-tracking branch 'origin/PlayingStateUI' into PlayingStateUI

# Conflicts:
#	MeesGame/Gameplay/UIObjects/Objects/Button.cs

maxmeijer42 committed:
Merge branch 'PlayingStateUI' of https://github.com/jensj12/MeesGame.git

Conflicts:
	MeesGame/Gameplay/GameStates/LoadMenuState.cs

maxmeijer42 committed:
Merge branch 'PlayingStateUI' of https://github.com/jensj12/MeesGame into PlayingStateUI

maxmeijer42 committed:
Merge branch 'PlayingStateUI' into loading-levels

Conflicts:
	MeesGame/Gameplay/GameStates/LoadMenuState.cs

maxmeijer42 committed:
Merge remote-tracking branch 'origin/master' into loading-levels

LisanneW committed:
keys dissapear after a player picks them up

maxmeijer42 committed:
Merge branch 'character' into demo

Conflicts:
	MeesGame/Gameplay/GameObjects/Player.cs
	MeesGame/Gameplay/GameObjects/PlayerState.cs
	MeesGame/Gameplay/GameObjects/TileField.cs

LisanneW committed:
Conflicts with master resolved

Commit:
Added buttons for saving and loading levels

LisanneW committed:
doors match surroundings

LisanneW committed:
keys are blue too

maxmeijer42 committed:
Merge remote-tracking branch 'origin/Keys' into demo

Conflicts:
	MeesGame/Gameplay/GameObjects/Player.cs
	MeesGame/Gameplay/GameObjects/TileField.cs

maxmeijer42 committed:
Merge branch 'demo' into BNijmeijer/FileIO-1.5

Conflicts:
	MeesGame/Gameplay/GameObjects/Player.cs
	MeesGame/Gameplay/GameObjects/PlayerState.cs
	MeesGame/Gameplay/GameObjects/Tile.cs
	MeesGame/Gameplay/GameObjects/TileField.cs
	MeesGame/Gameplay/Level/FileIO.cs
	MeesGame/Gameplay/Level/LevelData.cs
	MeesGame/MeesGame.csproj

jensj12 commented:
Merged via #87.
